iree/hal/dylib/cts/driver_registration.cc

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+#include "iree/hal/cts/cts_test_base.h"
+#include "iree/hal/dylib/registration/driver_module.h"
+
+namespace iree {
+namespace hal {
+namespace cts {
+
+iree_status_t register_test_driver(iree_hal_driver_registry_t* registry) {
```

Comment:
it'd be nice to avoid needing these files - they're just boilerplate overhead. Since the signature of all the registration methods is the same and you are stamping out c files anyway you can just extern the call in the template - just have the cts rule take the name of the registration function as an argument

Reply:
Yep, I can switch to generating these too. I wasn't quite sure about the right balance between autogenerated and allowing direct authoring. That suggestion would force a naming scheme on any drivers that want to use the CTS, while this allows some differences.

Reply:
I think passing the name of the registration function when declaring the suite is fine - and makes it much easier to search. Instead of a thousand characters here having 40 characters in the cmake file as a string seems like an acceptable tradeoff :)

Reply:
SG. I moved the registration into generated code.